python/lsst/dax/apdb/cassandra/apdbCassandra.py

```diff
@@ -565,10 +578,42 @@ def list_databases(cls, host: str) -> Iterable[str]:
         result = session.execute(query, (table_name,))
         keyspaces = [row[0] for row in result.all()]
 
+        infos = {keyspace: DatabaseInfo(name=keyspace) for keyspace in keyspaces}
+        if keyspaces:
+            # Retrieve roles for each keyspace.
+            template = ", ".join(["%s"] * len(keyspaces))
+            query = (
+                "SELECT resource, role, permissions FROM system_auth.role_permissions "
+                f"WHERE resource IN ({template}) ALLOW FILTERING"
+            )
+            resources = [f"data/{keyspace}" for keyspace in keyspaces]
+            try:
+                result = session.execute(query, resources)
+                for row in result:
+                    _, _, keyspace = row[0].partition("/")
+                    role: str = row[1]
+                    role_permissions: list[str] = list(row[2])
+                    info = infos[keyspace]
+                    if info.permissions is None:
+                        info.permissions = {role: role_permissions}
+                    else:
+                        info.permissions[role] = role_permissions
+            except cassandra.Unauthorized as exc:
+                # Likely that access to role_permissions is not granted for
+                # current user.
+                warnings.warn(
+                    f"Authentication information is not accessible to current user - {exc}", stacklevel=2
+                )
```

Comment:
Is it possible to say what account the query was run with? I can see new users not even realizing they're using credentials, let alone knowing which one.

Reply:
Account information is not available in this code, but the exception message from Cassandra does include role name, full warning will look like:
```
/sdf/data/rubin/.../dax_apdb/python/lsst/dax/apdb/scripts/list_cassandra.py:41: UserWarning: Authentication information is not accessible to current user - Error from server: code=2100 [Unauthorized] message="User apdb has no SELECT permission on <table system_auth.role_permissions> or any of its parents"
```